cl/scrapers/utils.py

```diff
@@ -334,43 +382,59 @@ def update_or_create_docket(
         "date_blocked": date_blocked,
         "date_argued": date_argued,
     }
-
-    docket = async_to_sync(find_docket_object)(
-        court_id, None, docket_number, None, None, None
-    )
-    if docket.pk:
-        # Update the existing docket with the new values
-        docket.add_opinions_source(source)
-
-        for field, value in docket_fields.items():
-            # do not use blanket `if not value:`, since
-            # blocked and ia_needs_upload are booleans and would be skipped
-            if value is None or value == "":
-                continue
-
-            if (
-                not overwrite_existing_data
-                and getattr(docket, field)
-                and getattr(docket, field) != value
-            ):
-                # Prevent overwriting values that already exist, since default values
-                # to this function are empty strings or None
-                logger.error(
-                    "Docket %s already has a %s %s, different than new value %s",
-                    docket.pk,
-                    field,
-                    getattr(docket, field),
-                    value,
-                )
-            else:
-                setattr(docket, field, value)
+    if from_harvard:
+        docket = async_to_sync(find_docket_object)(
+            court_id, None, docket_number, None, None, None
+        )
```

Comment:
Since I only know well the process for the scrapers, I prefer to leave the harvard process as it was, that's why it's separated, it's just keeping the old behavior. From what I have seen so far, sharing functions across different ingestion processes may silently introduce bugs

Apart from that, the `from_harvard` flags makes the updating overwrite all existing docket fields with the harvard data